pkg/controllers/jobset_controller.go

```diff
@@ -576,10 +574,9 @@ func constructJob(js *jobset.JobSet, rjob *jobset.ReplicatedJob, jobIdx int) (*b
 	labelAndAnnotateObject(job, js, rjob, jobIdx)
 	labelAndAnnotateObject(&job.Spec.Template, js, rjob, jobIdx)
 
-	// If enableDNSHostnames is set, update job spec to set subdomain as
-	// job name (a headless service with same name as job will be created later).
-	if dnsHostnamesEnabled(rjob) {
-		job.Spec.Template.Spec.Subdomain = GenSubdomain(js, rjob)
```

Comment:
this is the bug, we need to set the subdomain here

Reply:
`job.Spec.Template.Spec.Subdomain = js.Spec.Network.Subdomain`

Reply:
Ah you are right - I was too heavy handed in my deletion. I assumes that the webhook was setting this, but now I realize it's just ensuring that the subdomain is set on the original CRD spec!